### PR TITLE
optimize: the class registration method of kryo and fst

### PR DESCRIPTION
**Pull request by a364176773**
### Ⅰ. Describe what this PR did
optimize the class registration method of kryo and fst

### CODE REVIEW

File: serializer/seata-serializer-fst/src/main/java/io.seata.serializer.fst/FstSerializer.java [hunk 1]
```diff
@@ -26,16 +24,16 @@
 @LoadLevel(name = "FST")
 public class FstSerializer implements Serializer {
 
-    private final FSTConfiguration conf = FSTConfiguration.createDefaultConfiguration();
+    FstSerializerFactory fstFactory = FstSerializerFactory.getDefaultFactory();
```
Comment (slievrly) [inline serializer/seata-serializer-fst/src/main/java/io.seata.serializer.fst/FstSerializer.java]: private

File: core/src/main/java/io/seata/core/serializer/SerializerClassRegistry.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+    /**
+     * only supposed to be called at startup time
+     *
+     * @param clazz object type
+     * @param serializer object serializer
+     */
+    public static void registerClass(Class<?> clazz, Object serializer) {
+        if (clazz == null) {
+            throw new IllegalArgumentException("Class registered to kryo cannot be null!");
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/serializer/SerializerClassRegistry.java]: not only for kryo，where  used the  `serializer ` ?

Comment (a364176773) [reply core/src/main/java/io/seata/core/serializer/SerializerClassRegistry.java]: received, processed,thx

File: serializer/seata-serializer-fst/src/main/java/io.seata.serializer.fst/FstSerializer.java (hunk 1 above)
Comment (a364176773) [reply serializer/seata-serializer-fst/src/main/java/io.seata.serializer.fst/FstSerializer.java]: received, processed,thx

**Comment by slievrly** [comment]
will conflict with #2445

File: serializer/seata-serializer-fst/src/main/java/io.seata.serializer.fst/FstSerializerFactory.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * @author funkye
+ */
+public class FstSerializerFactory {
+
+    private static final FstSerializerFactory FACTORY = new FstSerializerFactory();
+
+    private final FSTConfiguration conf = FSTConfiguration.createDefaultConfiguration();
+
+
+    public static FstSerializerFactory getDefaultFactory() {
```
Comment (jsbxyyx) [inline serializer/seata-serializer-fst/src/main/java/io.seata.serializer.fst/FstSerializerFactory.java]: singleton? private constructor?

**Pull request merged**